src/drivers/dw/dma.c

```diff
@@ -289,8 +289,8 @@ static int dw_dma_start(struct dma_chan_data *channel)
 #if CONFIG_HW_LLI
 	/* LLP mode - write LLP pointer unless in scatter mode */
 	dma_reg_write(dma, DW_LLP(channel->index), lli->ctrl_lo &
-		      (DW_CTLL_LLP_D_EN | DW_CTLL_LLP_S_EN) ?
-		      (uint32_t)lli : 0);
+		      ((DW_CTLL_LLP_D_EN | DW_CTLL_LLP_S_EN) ?
+		       (uint32_t)lli : 0));
```

Comment:
@lgirdwood that's a possible source of errors. I have no idea what this code is supposed to do

Reply:
@lyakh Can you chime in on this one?

Reply:
Yeah, the unintended consequence of no precedence ;)
@ktrzcinx can you comment here since you have been working on this file recently. i.e. could this be working due to an error with precedence.

Reply:
I rather claim that this fix is incorect. With this additional () ctrl_lo doeesn't have impact on results, result is always lli, but as far as i know @tlauda is author of this line.

Reply:
no, that's definitely *not* what the original code was doing and most likely not what was intended. It isn't writing `ctrl_lo & x`, it is checkint `if (ctrl_lo & x)` and then writing either `lli` or 0. This hunk is breaking that.

Reply:
fixed, please double-check the update